client/packages/invoices/src/DoseQtyColumn.tsx

```diff
@@ -1,56 +1,62 @@
 import {
-  ArrayUtils,
   ColumnAlign,
   ColumnDefinition,
   RecordWithId,
   UNDEFINED_STRING_VALUE,
 } from '@openmsupply-client/common';
 import { ItemRowFragment } from '@openmsupply-client/system';
-import { InboundLineFragment } from './InboundShipment';
-import { StockOutLineFragment } from './StockOut';
+
+type VaccineItemRow = {
+  item: ItemRowFragment;
+  numberOfPacks: number;
+  packSize: number;
+  itemVariant?: { dosesPerUnit: number } | null;
+};
 
 export const getDosesQuantityColumn = <
-  T extends RecordWithId & {
-    item?: ItemRowFragment | null;
-    numberOfPacks?: number;
-    packSize?: number;
-  },
+  T extends RecordWithId & (VaccineItemRow | { lines: VaccineItemRow[] }),
 >(): ColumnDefinition<T> => ({
   key: 'doseQuantity',
   label: 'label.doses',
   width: 100,
   align: ColumnAlign.Right,
   accessor: ({ rowData }) => {
     if ('lines' in rowData) {
-      const { lines } = rowData as {
-        lines: (InboundLineFragment | StockOutLineFragment)[];
-      };
+      const { lines } = rowData;
+
       const isVaccine = lines[0]?.item?.isVaccine ?? false;
-      const unitQuantity = ArrayUtils.getUnitQuantity(lines);
+      const totalDoses = lines.reduce(
+        (sum, { packSize, numberOfPacks, item, itemVariant }) =>
+          sum +
+          packSize * numberOfPacks * (itemVariant?.dosesPerUnit ?? item.doses),
```

Comment:
oof, added a lot of helpers for this stuff in outbound allocation in doses PR, would be nice to reuse those once in...